tiledb/sm/array_schema/array_schema.h

```diff
@@ -338,6 +371,22 @@ class ArraySchema {
   Status add_attribute(
       shared_ptr<const Attribute> attr, bool check_special = true);
 
+  /**
+   * Adds an attribute with an enumeration
+   *
+   * @param attr The attribute to be added.
+   * @param enumeration The enumeration to set on the attribute.
+   * @param check_special If `true` this function will check if the attribute
+   *     is special (starting with `__`) and error if that's the case. Setting
```

Comment:
We didn't. I was just aping the existing `add_attribute` API in case that needed to be plumbed through. The new API design side steps the entire issue by not overloading `add_attribute` any more.